### PR TITLE
Feature/KYV-415 gdpr api

### PR DESCRIPTION
**Pull request by wpamb**

### CODE REVIEW

File: packages/order-experience-api/src/api/getGdpr.ts [hunk 1]
```diff
@@ -0,0 +1,451 @@
...
+import type { Response } from 'express'
+import * as yup from 'yup'
+import type { Order, Subscription } from '@verkkokauppa/order-backend'
+import {
+  getOrdersByUserAdmin,
+  getSubscriptionsByUserAdmin,
+} from '@verkkokauppa/order-backend'
+import { withAuthentication } from '@verkkokauppa/auth-helsinki-profile'
+
+interface Node {
```
Comment (severikupari1) [inline packages/order-experience-api/src/api/getGdpr.ts:18]: Should this interface be on different folder/file?

**Comment by severikupari1** [comment]
Did you write any automation tests for those filtering functions etc?

File: packages/order-experience-api/src/api/getGdpr.ts (hunk 1 above)
Comment (eerohakio) [reply packages/order-experience-api/src/api/getGdpr.ts:18]: Theres no need to always put interfaces to own files. Its good practise but not always needed if the interfaces are small and related only to specific place like in this case i guess.

**Review by eerohakio — APPROVED** [review]

File: packages/core/src/models/error.ts [hunk 2]
```diff
@@ -2,6 +2,7 @@ import type { Logger } from 'winston'
 
 export enum StatusCode {
   BadRequest = 400,
+  Unauthorized = 401,
```
[the remark this answers is not in the document]
Comment (wpamb) [reply packages/core/src/models/error.ts]: No, Unauthorized is correct

**Comment by wpamb** [comment]
> Did you write any automation tests for those filtering functions etc?

There's unit tests now

File: packages/order-experience-api/src/api/getGdpr.ts (hunk 1 above)
Comment (severikupari1) [reply packages/order-experience-api/src/api/getGdpr.ts:18]: Good point, I will remember that as well.

**Review by severikupari1 — APPROVED** [review]
Very good automation tests, thanks for those!

**Pull request merged**